### PR TITLE
✨: deepcopy: gen: match Go 1.17 build tag format

### PR DESCRIPTION
**Pull request by benpbrown**
The Go team has started the process of replacing the old "// +build"
constraints with the new "//go:build" constraints. See the design
doc for more details here:

https://go.googlesource.com/proposal/+/master/design/draft-gobuild.md

What is relevant is that in Go 1.17, gofmt now synchronizes the two
styles. In particular, if gofmt is run against a deepcopy generated
file, it will add a new "//go:build" constraint.

See https://golang.org/doc/go1.17#gofmt for details.

This PR intends to make it so that the generated code is compliant with
the new gofmt style.

Relevant issue: https://github.com/kubernetes-sigs/controller-tools/issues/594

### CODE REVIEW

**Comment by benpbrown** [comment]
I have signed the CLA.

**Comment by benpbrown** [comment]
Gentle ping for review @mengqiy @droot

**Comment by benpbrown** [comment]
Gentle ping for for review from the reviewers group (@droot @mengqiy @pwittrock @vincepri)

**Comment by benpbrown** [comment]
Ping for review @droot @mengqiy @pwittrock @vincepri

**Comment by benpbrown** [comment]
Ping for review @droot @mengqiy @pwittrock @vincepri

**Comment by camilamacedo86** [comment]
IHMO we need to update first k8s then controller runtime and then we can do this update here. See: https://github.com/operator-framework/operator-sdk/issues/5208#issuecomment-922272491

**Comment by benpbrown** [comment]
Hi @camilamacedo86 thanks for your message.

I understand that it would be desirable to upgrade the Go version used to compile Kubernetes (and related repos) all at once, but I don't quite follow the connection to this PR.

This PR doesn't use Go 1.17 for building controller-tools (it still uses 1.16). The change is in the codegen emitted by controller-tools, which becomes forwards compatible with 1.17.

Happily, the emitted code is backwards compatible with Go 1.16 ([Go 1.16 added support for the new style build tags](https://golang.org/doc/go1.16#go/build/constraint)), so contributors will continue to use Go 1.16 on this repo.

File: pkg/deepcopy/gen.go [hunk 1]
```diff
@@ -175,7 +175,8 @@ type ObjectGenCtx struct {
 // writeHeader writes out the build tag, package declaration, and imports
 func writeHeader(pkg *loader.Package, out io.Writer, packageName string, imports *importsList, headerText string) {
 	// NB(directxman12): blank line after build tags to distinguish them from comments
-	_, err := fmt.Fprintf(out, `// +build !ignore_autogenerated
+	_, err := fmt.Fprintf(out, `//go:build !ignore_autogenerated
+// +build !ignore_autogenerated
```
Comment (camilamacedo86) [inline pkg/deepcopy/gen.go:179]: Why do we need the old and the new syntax?

Comment (benpbrown) [reply pkg/deepcopy/gen.go:179]: The new syntax is not respected by previous Go versions (https://go.dev/doc/go1.16#go/build/constraint)

```
Note that //go:build lines are not supported in Go 1.16 and should not be introduced into Go programs yet.
```

**Review by camilamacedo86 — COMMENTED** [review]
/lgtm

**Review by alvaroaleman — APPROVED** [review]

**Comment by alvaroaleman** [comment]
/reopen
(to trigger the CLA bot)

**Pull request merged**